src/gentropy/l2g.py

```diff
@@ -205,89 +276,64 @@ def run_predict(self) -> None:
         ).parquet(self.predictions_path)
         self.session.logger.info("L2G predictions saved successfully.")
 
-    def run_train(self) -> None:
-        """Run the training step."""
-        if (
-            self.gs_curation
-            and self.interactions
-            and self.wandb_run_name
-            and self.model_path
-        ):
-            wandb_key = access_gcp_secret("wandb-key", "open-targets-genetics-dev")
-
-            # Instantiate classifier and train model
-            l2g_model = LocusToGeneModel(
-                model=GradientBoostingClassifier(random_state=42),
-                hyperparameters=self.hyperparameters,
-            )
-            wandb_login(key=wandb_key)
-            trained_model = LocusToGeneTrainer(
-                model=l2g_model,
-                feature_matrix=self._annotate_gold_standards_w_feature_matrix(),
-            ).train(self.wandb_run_name)
-            if trained_model.training_data and trained_model.model and self.model_path:
-                trained_model.save(self.model_path)
-                if self.hf_hub_repo_id and self.hf_model_commit_message:
-                    hf_hub_token = access_gcp_secret(
-                        "hfhub-key", "open-targets-genetics-dev"
-                    )
-                    trained_model.export_to_hugging_face_hub(
-                        # we upload the model in the filesystem
-                        self.model_path.split("/")[-1],
-                        hf_hub_token,
-                        data=trained_model.training_data._df.drop(
-                            "goldStandardSet", "geneId"
-                        ).toPandas(),
-                        repo_id=self.hf_hub_repo_id,
-                        commit_message=self.hf_model_commit_message,
-                    )
-
-    def _annotate_gold_standards_w_feature_matrix(self) -> L2GFeatureMatrix:
+    def run_train(self, gold_standard: L2GGoldStandard) -> None:
```

Comment:
I'd set `self.gold_standard` as an instance attribute.
```suggestion
    def run_train(self) -> None:
```

Reply:
I wanted to avoid setting `self.gold_standard = None` in the first place, because then we have to consecutively check if it is not None each time we want to refer to it (arbitrary code nesting that is not needed).

Given that the LocusToGeneStep `__init__` is actually performing the calculations, rather then just initialising, I could extract the logic to create a gold standard into the `__init__` rather then `run_train`. Then we can avoid the None checks downstream.